packages/xo-server/src/xo-mixins/xen-servers.js

```diff
@@ -472,6 +477,15 @@ export default class {
     const servers = await this._servers.get()
     const xapis = this._xapis
     forEach(servers, server => {
+      const lastEventFetchedTimestamp =
+        xapis[server.id]?.lastEventFetchedTimestamp
+      const currentTime = new Date().getTime()
+      if (
+        currentTime >
+        lastEventFetchedTimestamp + this._xapiMarkDisconnectedDelay
```

Comment:
What if `lastEventFetchedTimestamp` is `undefined`?

Reply:
`currentTime > Nan` =  'false' I've already tested this case

Reply:
I agree it works, but I think it would improve code clarity to test for `undefined` first.